### PR TITLE
feat: mobile site search for navbar #1559

### PR DESCRIPTION
**Pull request by mpleroux**
### 🔗 Linked issue

[CED-1950: Implement mobile site search navbar](https://energysage.atlassian.net/browse/CED-1950)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Refactored mobile navbar to accommodate new search button (magnifying glass)
- Restyled mobile search form to match design in Figma
- Added an ID prop to `EsSearchBar` to prevent duplicate IDs from multiple forms on the same page
- Refactored `EsSearchBar` to accommodate mobile version without breaking current desktop version
- Restyled mobile search form
- Prevented overlay element from covering mobile search form
- Added new ID prop to `EsSearchBar` to prevent duplicate form IDs
- Adjusted mobile navbar markup to use `showSearch` prop
- Handle `searchForm.focus()` for either mobile or desktop
- Hide mobile search form when mobile menus are opened
- Hide desktop search form when desktop menus are opened

### To Do

- [x] Hide search form when mobile menus are opened
    - Attempts to resolve this have resulted in circular logic
- [ ] Close mobile menus when opening mobile search form
    - See comments below for details

### 🥼 Testing

- Testing locally at:
    - http://localhost:8500/organisms/es-nav-bar
    - http://localhost:8500/organisms/es-nav-bar-with-search

#### 🧐 Feedback Requested / Focus Areas

- It may not be possible to use one `nav-search-bar` element for both mobile and desktop due to their locations in the navbar DOM

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

**Comment by hroth1994** [comment]
👀

**Comment by mpleroux** [comment]
> I see some CI failures on linting

I tried to fix that `<template>` tag error, but eslint or prettier keeps reformatting it as a single line. 

Thanks. I haven't been able to determine the cause of that error or other adjustments made to lines of code that I didn't modify. I believe Prettier is to blame, but the local editor configuration seems fine.

I disabled the VSCode extension in my workspace for that repo and manually reverted the changes. CI checks are now passing.

File: es-vue-base/src/lib-components/EsNavBar.vue [hunk 1]
```diff
@@ -300,28 +335,34 @@ export default {
...
+        const searchBarMobile = document.querySelector('.nav-search-bar-mobile');
         const productMenu = document.querySelector('.product-menu');
-        const searchIcon = document.querySelector('.search-icon');
-        const searchForm = document.getElementById('searchBar');
+        const searchIconMobile = document.querySelector('.search-icon-mobile');
+        const searchIconDesktop = document.querySelector('.search-icon-desktop');
+        const searchForm = document.getElementById('searchBarDesktop');
 
         // Function to show/hide search bar
         function toggle_search_bar(show_search_bar) {
```
Comment (ericdouglaspratt) [inline es-vue-base/src/lib-components/EsNavBar.vue:346]: Why not split this function into two, one for desktop and one for mobile? Then you would know which one to give focus to.

This function is already called from a specific click listener for the mobile search button, and a specific click listener for the desktop search button, so it should be straightforward to call the mobile or desktop versions of this function from those places, and call whatever the correct one is from all other places this function is called.

**Review by ericdouglaspratt — COMMENTED** [review]
What's the circular logic when you call `toggle_search_bar` from `show_mobile_menus`?

File: es-vue-base/src/lib-components/EsNavBar.vue (hunk 1 above)
Comment (mpleroux) [reply es-vue-base/src/lib-components/EsNavBar.vue]: That helped, thanks. The rest of the PR should be ready to review.

I was referring to one specific edge case when the user opens the mobile account menu then opens the mobile search menu. The menus overlap and it looks bad.

As far as I can tell, it would require pasting the contents of `collapse_mobile_menus()` and the functions it calls into `toggle_search_bar_mobile()`, which is defined first. It gets confusing.

I don't think this is critical enough to hold up the PR, but we should address it in the near future.

Comment (ericdouglaspratt) [reply es-vue-base/src/lib-components/EsNavBar.vue]: Looks fixed to me! If I open the search bar and then open either the account menu or the nav menu, the search bar closes. I'm not able to open an account/nav menu and then open the search bar, since those take up the whole screen, so I think we're fine there.

**Review by ericdouglaspratt — APPROVED** [review]
Looks good! Understood that we have a small edge case where wide mobile can open a menu and then the search bar, but for an A/B test, I don't think that's serious enough to warrant spending more time on.
